embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/CachedSession.kt

```diff
@@ -0,0 +1,71 @@
+package io.embrace.android.embracesdk.comms.delivery
+
+/**
+ * Represents a session that is cached on disk. The cached session encodes information in its
+ * filename such as sessionId, timestamp, and payload version. This allows the SDK to process
+ * cached sessions without deserializing the JSON.
+ */
+internal class CachedSession private constructor(
+    val sessionId: String,
+    val timestampMs: Long,
+    val filename: String,
+    val v2Payload: Boolean
+) {
+
+    companion object {
+        private const val V2_PREFIX = "v2"
+
+        /**
+         * Creates a [CachedSession] from a session id and timestamp.
+         */
+        fun create(sessionId: String, timestampMs: Long, v2Payload: Boolean): CachedSession {
+            val filename = when {
+                v2Payload -> getV2FileNameForSession(
+                    sessionId,
+                    timestampMs
+                )
+                else -> getV1FileNameForSession(sessionId, timestampMs)
+            }
+            return CachedSession(
+                sessionId,
+                timestampMs,
+                filename,
+                v2Payload
+            )
+        }
+
+        /**
+         * Creates a [CachedSession] from a filename.
+         */
+        fun fromFilename(filename: String): CachedSession? {
```

Comment:
Nice! Should we try to log an error or warning if we get a file we think is a session but somehow doesn't fit with the expected format?